### DISCUSSION TITLE
Setting values in RelationManager AttachAction

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#6909] [Help] [answered]
I have a `quotes` table and a `parts` table. They are joined in a many-many relationship in a `quote_parts` table.

In my `quotes` resource I have `LineItemsRelationManager` which I can successfully use to attach and detach parts. When using this, the user selects the part to be attached via a select component, enters a few more details, such as price and quantity, and the line item is added without a problem.

So far so good.

Now I want to add another attach action, but this time I want to set the part to be added without the user selecting it. So, for example, I want part_id to be 1 whenever this action is triggered.

This is the code:
```
AttachAction::make('Aggregate')
    ->form(fn (AttachAction $action): array => [
        $action->getRecordSelect(),
        TextInput::make('part_id')
            ->hidden()
            ->disabled()
            ->default(1),
        TextInput::make('group')
            ->required()
            ->default(1)
            ->numeric(),
        TextInput::make('discount')
            ->numeric(),
])
->mutateFormDataUsing(function (array $data): array {
    $data['part_id'] = 1;
    $data['price'] = 666;
    $data['quantity'] = 0;
    $data['group'] = 1;
    $data['sort'] = 1;
    return $data;
})
->after(
    function ($livewire) {
        $livewire->emit('refreshForm');
    }
),
```

Of course this isn't how I finally need it to work and I will be setting the values properly. The issue I'm having is that, because I'm using `getRecordSelect()` the user gets prompted for a part number. I've tried removing `$action->getRecordSelect(),` but then I get `Undefined array key "recordId"`.

I've tried setting `part_id` within the form as a hidden field as well as in `mutateFormDataUsing`. I guess I'm just not understanding something.

Any help with this gratefully received.

**Comment by pboivin** [comment, marked as answer]
I think the hidden field is problematic because it can be changed by a user in the browser's dev tools... Try removing `getRecordSelect()`, and setting the `recordId` key instead of `part_id` :
```php
->mutateFormDataUsing(function (array $data): array {
    $data['recordId'] = 1;
    // ...
```

**Reply by LemonPie22** [reply]
Thanks @pboivin. In the end I took another approach but your answer is appreciated and could well help people facing this issue in the future.
